src/SWP/Component/TemplatesSystem/Gimme/Widget/AbstractWidgetHandler.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: sourcefabric
+ * Date: 15/06/16
+ * Time: 16:15
+ */
+
+namespace SWP\Component\TemplatesSystem\Gimme\Widget;
+
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use SWP\Component\TemplatesSystem\Gimme\Model\WidgetModelInterface;
+
+abstract class AbstractWidgetHandler implements WidgetHandlerInterface, ContainerAwareInterface
+{
+    protected static $expectedParameters = array();
+
+    protected $container;
+
+    protected $widgetModel;
+
+    public static function getExpectedParameters()
+    {
+        return static::$expectedParameters;
+    }
+
+    public function __construct(WidgetModelInterface $widgetModel)
+    {
+        $this->widgetModel = $widgetModel;
+    }
+
+    public function setContainer(ContainerInterface $container = null)
+    {
+        $this->container = $container;
+    }
+
+    protected function getModelParameter($name)
+    {
+        if (array_key_exists($name, $this->widgetModel->getParameters())) {
```

Comment:
do you expect the value in `$this->widgetModel->getParameters()` array under the `$name` key to be null? - I don't think so. Because in this case if your parameter's value under the `$name` key will be null it will still return `true`. Then you should use `isset` instead which is also significantly faster from `array_key_exists` when running in many iterations.